modules/iot-core/README.md

```diff
@@ -0,0 +1,151 @@
+# Google Cloud IoT Core Module
+
+This module sets up Cloud IoT Core Registry, registers IoT Devices and configures Pub/Sub topics required in Cloud IoT Core.
+
+To use this module, ensure the following APIs are enabled:
+* pubsub.googleapis.com
+* cloudiot.googleapis.com
+
+## Simple Example
+
+Basic example showing how to create an IoT Platform (IoT Core), connected to a set of given Pub/Sub topics and provision IoT devices.
+
+Devices certificates must exist before calling this module. You can generate these certificates using the following command
+
+```
+openssl req -x509 -newkey rsa:2048 -keyout rsa_private.pem -nodes -out rsa_cert.pem -subj "/CN=unused"
+```
+
+And then provision public certificate path, together with the rest of device configuration in a devices yaml file following the following format
+```
+device_id: # id of your IoT Device
+is_blocked: # false to allow device connection with IoT Registry
+is_gateway: # true to indicate the device connecting acts as a gateway for other IoT Devices
+log_level: # device logs level
+certificate_file: # public certificate path, generated as explained in the previous step
+certificate_format: # Certificates format values are RSA_PEM, RSA_X509_PEM, ES256_PEM, and ES256_X509_PEM
+```
+
+Example Device config yaml configuration
+```
+device_1:
+is_blocked: false
+is_gateway: false
+log_level: INFO
+certificate_file: device_certs/rsa_cert5.pem
+certificate_format: RSA_X509_PEM
+device_2:
+is_blocked: true
+is_gateway: false
+log_level: INFO
+certificate_file: device_certs/rsa_cert5.pem
+certificate_format: RSA_X509_PEM
+```
```

Comment:
yamldecode() of this example returns
```hcl
{
  "certificate_file" = "device_certs/rsa_cert5.pem"
  "certificate_format" = "RSA_X509_PEM"
  "device_1" = null
  "device_2" = null
  "is_blocked" = true
  "is_gateway" = false
  "log_level" = "INFO"
}
```
Is the example formatted correctly?

I think you want
```yaml
device_1:
  is_blocked: false
  is_gateway: false
  log_level: INFO
  certificate_file: device_certs/rsa_cert5.pem
  certificate_format: RSA_X509_PEM
device_2:
  is_blocked: true
  is_gateway: false
  log_level: INFO
  certificate_file: device_certsrsa_cert5.pem
  certificate_format: RSA_X509_PEM
```